python/tank/util/local_file_storage.py

```diff
@@ -164,6 +164,10 @@ def get_site_root(cls, hostname, path_type, generation=CORE_V18):
                            which is the current generation of paths.
         :return: Path as string
         """
+        if hostname is None:
```

Comment:
Why do you feel this should be a tank error? It's input validation and we have determined that a value that is passed to the method isn't within the valid range of input values?